.evergreen/compile-only.sh

```diff
@@ -6,9 +6,25 @@ source ./.evergreen/configure-rust.sh
 rustup update $RUST_VERSION
 
```

Comment:
since the runtime-specific compile scripts were only used from this file, I took the opportunity to cut out two scripts and just move their logic into here.